packages/program-boilerplate/README.md

```diff
@@ -1,17 +1,226 @@
+`<my-program>.ts` should be the entry point for running the program on Heroku that starts
+the Express server. You can take a look at the referral program for examples if you are
+unsure.
+
+The rest of the program structure is a matter of personal taste; take a look at the
+referral program for an example to follow if you are unsure.
+
+To get your program started, fill the source files with some boilerplate:
+
+`program.ts`
+```typescript
+import {Program} from '@saasquatch/program-boilerplate';
+
+export const program: Program = {
+  AFTER_USER_CREATED_OR_UPDATEDL: undefined,
+  AFTER_USER_EVENT_PROCESSED: undefined,
+  REFERRAL: undefined,
+  PROGRAM_INTROSPECTION: undefined,
+  SCHEDULED: undefined,
+  REWARD_SCHEDULED: undefined,
+  PROGRAM_VALIDATION: undefined,
+};
+```
+
+`<my-program>.ts`
+```typescript
+import {program} from './program';
+import {webtask} from '@saasquatch/program-boilerplate';
+
+webtask(program).listen(process.env.PORT, () => {
+  console.log(`My program running on port ${process.env.PORT}`);
+});
+```
+
+Running `npm build && npm start` should now run your program locally (useless for now,
+but a good test to make sure things are working).
+
+## Set up your program template
+As mentioned previously, your program also needs a template which defines its rules,
+rewards, emails etc. You could try to write your template from scratch, but it would
+likely be easier to copy/paste one from another program and edit it accordingly. You can
+probably skip this step and come back to it later.
+
+## Provisioning cloud infrastructure
+Next it is time to provision your cloud infrastructure that will host your program
+components.
+
+### Creating a Heroku pipeline
+In the Heroku console, switch to the `saasquatch-webtasks` team. Select `New` -> `Create
+new pipeline`. The name of the pipeline should match the name for your program that was
+chosen in the previous step. Do not connect to GitHub.
+
+Next, add the staging and prod apps to the pipeline by clicking `Add app` -> `Create new
+app...` under the respective staging and production pipeline stages. By convention, the
+app names are `<my-program>-staging` and `<my-program>-prod`.
+
+#### Setting up the Heroku app
+Now that your apps are created, you can configure them to run your program. It's easiest
+to do this from the command line. The first thing you will want to do is use the
+`heroku.sh` script to set up the git remotes.
+
+From the `programs` folder, run:
+```
+./heroku.sh setup-remotes
+```
+This will set up the git remotes for making deployments to the various programs. It's OK
+if you get a bunch of `fatal: remote <some_remote> already exists.` errors.
+
+**WARNING**:
+The `setup-remotes` command will only work if the folder name for your program matches
+the names of the Heroku programs, ie. you have a folder: `my-program` and Heroku programs
+`my-program-staging` and `my-program-prod`.
+
```

Comment:
There's probably another step to setup SSL but only if you're going to run the program on a custom domain.